### PR TITLE
Roadworks

### PR DESCRIPTION
**Pull request by leonardehrenfried**
Adds a roadworks tile layer with a popup which looks like this:

Screenshot from 2019-12-06 15-42-53

### CODE REVIEW

File: app/component/map/popups/RoadworksPopup.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import Relay from 'react-relay/classic';
+import moment from 'moment';
+import Card from '../../Card';
+import CardHeader from '../../CardHeader';
+import { station as exampleStation } from '../../ExampleData';
```
Comment (hbruch) [inline app/component/map/popups/RoadworksPopup.js]: What is that needed for?

File: app/component/map/popups/RoadworksPopup.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  static contextTypes = {
+    getStore: PropTypes.func.isRequired,
+  };
+
+  static description = (
+    <div>
+      <p>Renders a roadworks popup.</p>
+      <ComponentUsageExample description="">
+        <RoadworksPopup context="context object here" station={exampleStation}>
+          Im content of a roadworks card
```
Comment (hbruch) [inline app/component/map/popups/RoadworksPopup.js]: What is the exampleStation needed for?

File: app/component/map/tile-layer/Roadworks.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+      }
+
+      return res.arrayBuffer().then(
+        buf => {
+          const vt = new VectorTile(new Protobuf(buf));
+
+          this.features = [];
+
+          // it's called "roadwork" in the tile source but i want to change it to "roadworks" since that is more
+          // consistent
```
Comment (hbruch) [inline app/component/map/tile-layer/Roadworks.js]: So this is only for backward compatibility? Could you do all changes so we don't need this any longer?

File: app/component/map/tile-layer/Roadworks.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+      );
+    });
+
+  fetchAndDrawStatus = ({ geom, properties }) => {
+    console.log('Drawing roadworks icon', geom, properties);
+    var suffix = '';
+    if (properties.Vollsperrung === 1) {
+      suffix = '-full-closure';
+    } else if (properties.Halbseitige_Sperrung === 1) {
+      suffix = '-partial-closure';
```
Comment (hbruch) [inline app/component/map/tile-layer/Roadworks.js]: -oneway-closure would be describe it better

File: app/component/map/tile-layer/VectorTileLayerContainer.js [hunk 5]
```diff
@@ -26,8 +27,12 @@ export default function VectorTileLayerContainer(props, { config }) {
       layers.push(TicketSales);
     }
 
-    if (config.dynamicParkingLots && config.dynamicParkingLots.showDynamicParkingLots) {
-      layers.push(DynamicParkingLots)
+    if (config.constructor && config.dynamicParkingLots.showDynamicParkingLots) {
+      layers.push(DynamicParkingLots);
+    }
+
+    if (config.constructor && config.roadworks.showRoadworks) {
```
Comment (hbruch) [inline app/component/map/tile-layer/VectorTileLayerContainer.js]: Above, the existence of config.<layer> was checked before accessing the show<Layer> property. Could you explain why config.constructor makes this unnecessary? What would happen for e.g. a HSL config if it roadworks is not defined at all?

File: static/assets/svg-sprite.default.svg [hunk 6]
```diff
@@ -1010,5 +1010,81 @@
...
+			<path id="path3650" style="fill:#000000;fill-opacity:1;fill-rule:nonzero;stroke:none" d="m 23.859,456.511 0,-0.047 -0.05,0 0.05,0.047 z"/>
+			<path id="path3652" style="fill:#000000;fill-opacity:1;fill-rule:nonzero;stroke:none" d="m 23.809,456.511 0.051,0 0,-0.047 -0.051,0 0,0.047 z"/>
+			<path id="path3654" style="fill:#000000;fill-opacity:1;fill-rule:nonzero;stroke:none" d="m 23.809,456.464 0,0.047 0.05,-0.047 -0.05,0 z m 0.05,0 -0.05,0.047 0,-0.047 0.05,0 z"/>
+			<path id="path3656" style="fill:#000000;fill-opacity:1;fill-rule:nonzero;stroke:none" d="m 23.859,456.464 -0.05,0 0.023,0 0,0.047 0.027,-0.047 z"/>
+			<path id="path3658" style="fill:#000000;fill-opacity:1;fill-rule:nonzero;stroke:none" d="m 23.809,456.464 0,0.047 0,-0.047 0.05,0.047 -0.05,-0.047 z"/>
+			<path id="path3660" style="fill:#000000;fill-opacity:1;fill-rule:nonzero;stroke:none" d="m 23.809,456.511 0.051,0 0,-0.047 -0.051,0 0,0.047 z"/>
+		</g>
+	</g>
+</symbol>
+<symbol id="icon-icon_roadworks-partial-closure" viewBox="0 0 600.99628 600.99628" height="600.99628" width="600.99628">
```
Comment (hbruch) [inline static/assets/svg-sprite.default.svg]: As said above: oneway-closure describes it better

File: app/component/map/tile-layer/Roadworks.js (hunk 3 above)
Comment (leonardehrenfried) [reply app/component/map/tile-layer/Roadworks.js]: I forgot to push! https://github.com/mfdz/tilelive-roadworks-bw/commit/1ca481244aa5436a036e50cdb5b4fdbcb91ef90a

File: app/component/map/popups/RoadworksPopup.js (hunk 1 above)
Comment (leonardehrenfried) [reply app/component/map/popups/RoadworksPopup.js]: A leftover from a copy&past josb. I removed it.

File: app/component/map/popups/RoadworksPopup.js (hunk 2 above)
Comment (leonardehrenfried) [reply app/component/map/popups/RoadworksPopup.js]: Not needed. I removed it.

File: static/assets/svg-sprite.default.svg (hunk 6 above)
Comment (leonardehrenfried) [reply static/assets/svg-sprite.default.svg]: Changed.

File: app/component/map/tile-layer/VectorTileLayerContainer.js (hunk 5 above)
Comment (leonardehrenfried) [reply app/component/map/tile-layer/VectorTileLayerContainer.js]: To be honest, I have no idea why I put this here. I suspect a copy'n paste from somewhere else.

I changed it.

File: app/component/map/tile-layer/Roadworks.js (hunk 4 above)
Comment (leonardehrenfried) [reply app/component/map/tile-layer/Roadworks.js]: Changed.

**Comment by leonardehrenfried** [comment]
Most of the eslint/prettier errors were easy to fix but there is a newish JS feature I was unfamiliar with: https://github.com/mfdz/digitransit-ui/pull/36/commits/fb9dfd2b8d6353b4e90da2cd34d74f64d81ae0c5#diff-a698d8540a260c6252893d3a23835a02R54-R56

The build would not pass without this change.

Since I sadly don't have the time to spin up the complete stack to try this out but can do so on Friday. Alternatively, you can build the containers and try it out yourself.

**Comment by leonardehrenfried** [comment]
The tests failed but it seems unrelated to what I did.

**Comment by leonardehrenfried** [comment]
Please also check out #39 first. I fixed the failing unit tests there.

**Comment by hbruch** [comment]
Could you merge master into your branch which should fix the failing tests, thx.

**Pull request merged**